src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpServer.cs

```diff
@@ -77,12 +119,17 @@ private static NativeMethods.REQUEST_NOTIFICATION_STATUS HandleRequest(IntPtr pI
         {
             // Unwrap the server so we can create an http context and process the request
             var server = (IISHttpServer)GCHandle.FromIntPtr(pvRequestContext).Target;
+            if (server != null)
```

Comment:
Why bother checking for null here? The next line immediately dereferences server